packages/stokhos/src/sacado/kokkos/pce/Sacado_UQ_PCE_Imp.hpp

```diff
@@ -392,7 +392,7 @@ operator*=(const PCE<Storage>& x)
   const ordinal_type sz = this->size();
   const ordinal_type csz = sz > xsz ? sz : xsz;
 
-#if !defined(__CUDA_ARCH__)
+#if !(defined(__CUDA_ARCH__) || defined(__HIP_DEVICE_COMPILE__))
```

Comment:
All these should probably be `KOKKOS_IF_ON_HOST((..))` for portability.

Reply:
Fixed